src/modules/commander/Commander.cpp

```diff
@@ -1190,49 +1189,125 @@ bool
 Commander::set_home_position()
 {
 	// Need global and local position fix to be able to set home
-	if (status_flags.condition_global_position_valid && status_flags.condition_local_position_valid) {
+	// but already set the home position in local coordinates if available
+	// in case the global position is only valid after takeoff
+	if (status_flags.condition_local_position_valid) {
+
+		// Set home position in local coordinates
+		const vehicle_local_position_s &lpos = _local_position_sub.get();
+		_heading_reset_counter = lpos.heading_reset_counter;
+
+		home_position_s home{};
+		home.timestamp = hrt_absolute_time();
+		home.manual_home = false;
+		fillLocalHomePos(home, lpos);
+
+		if (status_flags.condition_global_position_valid) {
+
+			const vehicle_global_position_s &gpos = _global_position_sub.get();
+
+			// Ensure that the GPS accuracy is good enough for intializing home
+			if (isGPosGoodForInitializingHomePos(gpos)) {
+				fillGlobalHomePos(home, gpos);
+				setHomePosValid();
+			}
+		}
+
+		_home_pub.update(home);
```

Comment:
We'll need to verify all the consumes of `home_position` are checking the relevant flags if we're going to publish before it's overall considered valid (condition_home_position_valid).

Reply:
addressed in d8565d7